### PR TITLE
[BI-1230] Truly Unknown Germplasm

### PR DESCRIPTION
**Pull request by HMS17**
# Description
**Story:** [BI-1230 - Truly Unknown Germplasm](https://breedinginsight.atlassian.net/browse/BI-1230)

Updates to support importing germplasm with truly unknown parents without using a dummy parent. This development also ensures that unrelated germplasm with unknown parents are not falsely attributed as siblings in the pedigree view display.

To import a file with unknown parents, a GID or entry number of 0 must be used. Presently only both parents unknown or female parent known and male parent unknown are supported. 

Unknown parents will be displayed as "Unknown" in the parent GID columns in the All Germplasm table and in the pedigree in Germplasm Details page.

Due to Breedbase limitations, parents with more than 10 progeny will cause the pedigree viewer to error (to be fixed in BI-1579.

# Dependencies
[bi-api/BI-1230](https://github.com/Breeding-Insight/bi-api/pull/205)

# Testing
For both import with parent entry numbers and import with germplasm gids, import germplasm file with:

- Germplasm with no parents (should not be labeled unknown)
- Germplasm with unknown female parent
- Germplasm with known female parent and unknown male parent
- Germplasm with unknown female parent and unknown male parent 

Check import occurs successfully
Check that Unknown is displayed in GID column for All Germplasm Table for germplasm with Unknown parent
Check that Unknown displayed in pedigree on Germplasm Details page for germplasm with Unknown parent

Look at pedigree viewer and expand generations, unknown parents should be displayed and be separate nodes from each other

# Checklist:

- [X] I have performed a self-review of my own code
- [X] I have tested my code and ensured it meets the acceptance criteria of the story
- [ ] I have create/modified unit tests to cover this change
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to documentation
- [ ] I have run TAF: [_\<link to TAF run>_](https://github.com/Breeding-Insight/taf/actions/runs/3009574983)

### CODE REVIEW

File: src/breeding-insight/model/import/germplasm/Pedigree.ts [hunk 1]
```diff
@@ -44,4 +44,18 @@ export class Pedigree {
...
   }
 
+  /**
+   * Wrapper of parsePedigreeString that handles female/male parent unknown cases.
+   *
+   * @param pedigreeString
+   * @param femaleParentUnknown
+   * @param maleParentUnknown
+   */
+  public static parsePedigreeStringWithUnknowns(pedigreeString: string, femaleParentUnknown: string, maleParentUnknown: string, gid: string) : Pedigree {
```
Comment (timparsons) [inline src/breeding-insight/model/import/germplasm/Pedigree.ts:54]: It looks like the `femaleParentUnknown` and `maleParentUnknown` additionalInfo fields are booleans.  Can the types of the parameters in this method can be updated to reflect this?

**Review by timparsons — CHANGES_REQUESTED** [review]
Dev testing passed.  Code looks good with one minor change request

File: src/breeding-insight/model/import/germplasm/Pedigree.ts (hunk 1 above)
Comment (HMS17) [reply src/breeding-insight/model/import/germplasm/Pedigree.ts:54]: Vue yells at me if I change the parameter type to boolean despite it being very clearly processed as a boolean in the method. It is strange.

Comment (timparsons) [reply src/breeding-insight/model/import/germplasm/Pedigree.ts:54]: Ok, thanks for looking into this!

**Review by timparsons — APPROVED** [review]

**Review by nickpalladino — APPROVED** [review]

**Pull request merged**